src/blocks/github.rs

```diff
@@ -0,0 +1,277 @@
+extern crate lazy_static;
+
+use crate::blocks::Update;
+use crate::blocks::{Block, ConfigBlock};
+use crate::config::Config;
+use crate::de::deserialize_duration;
+use crate::errors::*;
+use crate::input::I3BarEvent;
+use crate::scheduler::Task;
+use crate::util::FormatTemplate;
+use crate::widget::I3BarWidget;
+use crate::widgets::text::TextWidget;
+use crossbeam_channel::Sender;
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde_derive::Deserialize;
+use std::collections::HashMap;
+use std::process::Command;
+use std::time::Duration;
+use uuid::Uuid;
+
+const GITHUB_TOKEN_ENV: &str = "GITHUB_TOKEN";
+
+pub struct Github {
+    text: TextWidget,
+    id: String,
+    update_interval: Duration,
+    api_server: String,
+    token: String,
+    format: FormatTemplate,
+}
+
+#[derive(Deserialize, Debug, Default, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct GithubConfig {
+    /// Update interval in seconds
+    #[serde(
+        default = "GithubConfig::default_interval",
+        deserialize_with = "deserialize_duration"
+    )]
+    pub interval: Duration,
+
+    #[serde(default = "GithubConfig::default_api_server")]
+    pub api_server: String,
+
+    /// Format override
+    #[serde(default = "GithubConfig::default_format")]
+    pub format: String,
+}
+
+impl GithubConfig {
+    fn default_interval() -> Duration {
+        Duration::from_secs(30)
+    }
+
+    fn default_api_server() -> String {
+        "https://api.github.com".to_owned()
+    }
+
+    fn default_format() -> String {
+        "{total}".to_owned()
+    }
+}
+
+impl ConfigBlock for Github {
+    type Config = GithubConfig;
+
+    fn new(block_config: Self::Config, config: Config, _: Sender<Task>) -> Result<Self> {
+        let token = match std::env::var(GITHUB_TOKEN_ENV).ok() {
```

Comment:
It seems like GITHUB_TOKEN might be used by some official GitHub tools: https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#about-the-github_token-secret

Do you think it's better to rename the env to something like I3RS_GITHUB_TOKEN?

Reply:
Yeah it makes sense. If one wants to use one different token per app, that's a valid use case.  Will change to `I3RS_GITHUB_TOKEN`.

Reply:
Yeah it makes sense. If one wants to use one different token per app, that's a valid use case.  Will change to `I3RS_GITHUB_TOKEN`.